ee/tabby-webserver/src/service/utils/prompt.rs

```diff
@@ -103,17 +103,32 @@ pub fn transform_line_items(content: &str) -> Vec<String> {
         .lines()
         .map(trim_bullet)
         .filter(|x| !x.is_empty())
+        .map(|x| x.to_owned())
         .collect()
 }
 
 /// Trims leading and trailing bullet-like characters or digits from the provided string and returns the trimmed result.
-pub fn trim_bullet(s: &str) -> String {
-    let is_bullet = |c: char| c == '-' || c == '*' || c == '.' || c.is_numeric();
-    s.trim()
-        .trim_start_matches(is_bullet)
-        .trim_end_matches(is_bullet)
-        .trim()
-        .to_owned()
+fn trim_bullet(s: &str) -> &str {
+    let s = s.trim();
+    if s.starts_with('-') || s.starts_with('+') || s.starts_with('*') {
+        return s.trim_matches(&['-', '+', '*', ' '][..]);
+    }
+
+    // Check for numbered list
+    if let Some(pos) = s.find(". ") {
+        let prefix = &s[..pos];
+        if prefix.chars().all(|c| c.is_numeric()) {
+            return trim_bullet(s[pos + 2..].trim_start());
+        }
+    }
+
+    s
+}
+
+pub fn trim_title(title: &str) -> &str {
+    // take first line.
+    let title = title.lines().next().unwrap_or(title).trim();
+    trim_bullet(title.trim_matches(&['"', '#', ' ', '-', '*'][..]))
```

Comment:
```suggestion
   title.trim_matches(&['"', '#', ' ', '-', '*'][..])
```

prefer not nesting trimming rules.